### ISSUE TITLE
The use of typeswitch feature flag needs to be deleted

### ISSUE THREAD

**Issue by jaredpar** [dotnet/roslyn#12537]
C# 7.0 language features in the master branch should not have behavior tied to a feature switch.  This particular expression is doing so:

``` C#
(parseOptions?.Features.ContainsKey("typeswitch") == true ||
```

Source: https://github.com/dotnet/roslyn/blob/master/src/Compilers/CSharp/Portable/Binder/SwitchBinder_BindPatternSwitch.cs#L39

**Comment by gafter** [comment]
This flag is present for test coverage and induces no semantic changes. If I remove the flag we'll lose that coverage, but if that is what you prefer...

**Comment by gafter** [comment]
You assigned this to preview 4. You want us to go into ask mode to reduce test coverage? I don't understand the urgency.

**Comment by jaredpar** [comment]
Feature flags should not be used for test coverage. The coverage should be established using code users will be typing and compiling.

I meant for this to be RV, not preview 4

Get Outlook for iOS<https://aka.ms/o0ukef>




On Thu, Jul 14, 2016 at 4:38 PM -0700, "Neal Gafter" <notifications@github.com<mailto:notifications@github.com>> wrote:


This flag is present for test coverage and induces no semantic changes. If I remove the flag we'll lose that coverage, but if that is what you prefer...

-

**Comment by AdamSpeight2008** [comment]
Can we have some clarity on feature flag usage? 
Or is @jaredpar 's comment about directly checking the `Features` dictionary? 
It should be noted there is a PR in flight (for vb) that moves the check into `CheckFeatureAvailability` function. I think C# should also consider doing the same.

**Comment by gafter** [comment]
The feature flag is present to support deeper unit testing, not to change the language behavior. That is one of the reasons the feature flag machinery was added to the compilers. The unit tests are valuable, so we should keep them in our suite.

**Comment by AdamSpeight2008** [comment]
Shouldn't the feature flag text (eg `("typeswitch")` ) be gotten via the function  
[GetFeatureFlag](https://github.com/dotnet/roslyn/blob/dcfb3f9c83a77dad7a8a721d7b8413e0dd243d34/src/Compilers/VisualBasic/Portable/Parser/ParserFeature.vb#L38)  or the equivalent in C#? So that the is only one site where the literal string `"typeswitch"` is defined.
